File: processingjs/js/sf4_loops.js
"use strict";

function sketch4(p) {

	let x = 100;
	let y = 100;
	let s = 50;
	let speed = 5;

	p.setup = function() {
		p.size(640, 360);
	}

	p.draw = function() {
		p.background(127);

	    for (let i=0; i<4; i++) {
	    	p.ellipse(x, y + (i * s), s, s);
	    }

  		x += speed;
	  
		if (x > p.width || x < 0) {
			speed *= -1;
		}
	}

}

File: p5py/sf4_loops.py
from p5 import *

x = 100
y = 100
s = 50
speed = 5

def setup():
    size(640, 360)

def draw():
    global x, y, s, speed
    background(127)
    
    for i in range(0, 4):
        ellipse(x, y + (i * s), s, s)
    
    x += speed
    
    if x > width or x < 0:
        speed *= -1

run()